src/sql/parts/notebook/notebookActions.ts

```diff
@@ -137,11 +142,124 @@ export class KernelsDropdown extends SelectBox {
 }
 
 export class AttachToDropdown extends SelectBox {
-	constructor(container: HTMLElement, contextViewProvider: IContextViewProvider) {
-		let selectBoxOptionsWithLabel: ISelectBoxOptionsWithLabel = {
-			labelText: attachToLabel,
-			labelOnTop: false
-		};
-		super([msgLocalHost], msgLocalHost, contextViewProvider, container, selectBoxOptionsWithLabel);
+	private model: INotebookModel;
+
+	constructor(container: HTMLElement, contextViewProvider: IContextViewProvider, modelRegistered: Promise<INotebookModel>,
+		@IConnectionManagementService private _connectionManagementService: IConnectionManagementService,
+		@IConnectionDialogService private _connectionDialogService: IConnectionDialogService,
+		@INotificationService private _notificationService: INotificationService) {
+		super([msgLoadingContexts], msgLoadingContexts, contextViewProvider, container, { labelText: attachToLabel, labelOnTop: false} as ISelectBoxOptionsWithLabel);
+		if (modelRegistered) {
+			modelRegistered
+			.then((model) => this.updateModel(model))
+			.catch((err) => {
+				// No-op for now
+			});
+		}
+		this.onDidSelect(e => {
+			let connection = this.model.contexts.otherConnections.find((c) => c.options.host === e.selected);
+			this.doChangeContext(connection);
+		});
+	}
+
+	public updateModel(model: INotebookModel): void {
+		this.model = model;
```

Comment:
I'm really looking forward to all this logic moving back out to an extension / back to an object with defined callbacks, instead of inheriting from selectbox and interacting with raw HTMLElements. Not in this PR, but soon I hope?

Reply:
Yes, definitely looking forward to doing that. I've started thinking about the APIs that we'll need. Let's have a sync about this next week.